src/flows/pipes/Notification/index.ts

```diff
@@ -1,5 +1,10 @@
 import View from './view'
 import ReadOnly from './readOnly'
+import {PipeData} from 'src/types'
+
+const removeNotificationThresholds = data => {
+  delete data.thresholds
```

Comment:
this is not what that function is for. what are you trying to do here? the next step in the chain is that `data` gets deleted,  so  i'm unsure the purpose of removing a field from it before removing the whole object

Reply:
This error was occurring:
https://github.com/influxdata/ui/issues/4479

That exact change, fixed the error.
It was occurring for the javascript pointer object (the array), but not the primitive (measurement).

Reply:
please remove this improperly added memo instead. it's preventing the threshold allocation from being properly recycled with garbage collection
https://github.com/influxdata/ui/blob/master/src/flows/pipes/Notification/Threshold.tsx#L40

Reply:
It's not the memoize. (That component is completely unmounted.)

Tested:
1. reverted the commit with the `removeNotificationThresholds()`
2. removed the memo.
  * <img width="350" alt="Screen Shot 2022-04-26 at 10 01 34 AM" src="https://user-images.githubusercontent.com/10232835/165354116-059c9854-dc67-45f1-83a5-344b3ff40577.png">

3. the issue/bug is still present (without the memo).


https://user-images.githubusercontent.com/10232835/165354223-7a3c24a8-bbef-47af-be1d-3b03ea81d0ad.mov

Reply:
does that mean that we're not properly deep cloning from the `initial` state?

Reply:
agreed. it's a pass by reference error somewhere.
I was forcing deletion of the reference, to prevent this error.

We could alternatively enforce a deep cloning of the init state whenever we addPipe. I like that idea. Will do.